client/init/proximity.lua

```diff
@@ -113,10 +118,9 @@ CreateThread(function()
 		if voiceState == "proximity" then
 			addNearbyPlayers()
 			local isSpectating = NetworkIsInSpectatorMode()
-			if isSpectating and not isListenerEnabled then
-				setSpectatorMode(true)
-			elseif not isSpectating and isListenerEnabled then
-				setSpectatorMode(false)
+			if wasSpectating ~= isSpectating then
+				wasSpectating = isSpectating
+				setSpectatorMode(isSpectating)
```

Comment:
Unless I'm looking at this wrong this won't do anything? If they're not in spectator mode it will still reset them out of spectator mode, you need to have some sort of override for listener mode

Reply:
It might also be more intuitive use `GetRenderingCam` and if its not -1 to set this

Reply:
I agree about GetRenderingCam change, that could automate this and be more intuitive

However this commit is correct, if they are not in spectator mode and override spectator mode, this will not reset them out
But, switch on then off the spec mode would reset the override